subnet/subnet.go

```diff
@@ -25,7 +25,8 @@ import (
 )
 
 type LeaseAttrs struct {
-	PublicIP    ip.IP4
+	InterfaceIP ip.IP4
```

Comment:
After thinking about this, I'd prefer not to serialize `InterfaceIP` and use `PublicIP` for identity. Here're are the reasons:

1. Like you said, you can get two machines with the same private IP.
2. If one re-associates public IP from one machine to another, they can both get the same subnet. But they'll need to explicitly change the `--public-ip` cmd arg and should remember to shut down the other node (or something). Even if they don't, the traffic will still only flow to the new node.
3. Nothing changes in etcd. We can always add more fields later but for now everything is trivially forward and backward compatible.